# Тестовое задание на вакансию «Аналитик маркетплейсов»

подготовил: [Вова Сизов](mailto:vladimirsizov@yandex.ru)

<hr>
Задача: Вам предоставлено 4 сводных таблицы.   
Основной вопрос: Сколько заработано чистой прибыли по следующим товарам?

1.Мотоперчатки именно мотоперчатки (не зимние, не тактические, не fox, просто мотоперчатки)  
2.Мотоштаны  
3.Мотокуртка   
4.Мотошлем  
5.Мототент  


Комментарии к таблицам:

1.Первая таблица – Аспо детализация  
это файл, который присылает нам ваилдберриз (в нем указаны расходы на логистику, расходы, комиссии, штрафы и так далее). Разбивать товары по размерам не надо, но надо разбить по цветам, например «мотокуртка красная», «мотокуртка черная». 
Эта таблица помогает нам понять, какие были расходы, чтобы посчитать чистую прибыть (валовая выручка после комиссии маркетплейса)

2.Таблица «Фин модель»   
отсюда мы получаем информацию о себестоимости (закупка+логистика из Китая). Себестоимость может меняться, так как скачет курс, это нужно учитывать при заполнении ячейки таблицы

3.Таблица «Расходы на рекламу»  
также по каждому артиклю (разбиваем только по цвету!) смотрим, сколько потрачено на рекламу

4.Таблица «Отчет по платному хранению»   
отсюда берем информацию о том, сколько потратили на хранение (он, к сожалению, не разбивается по артиклям, надо понять, как отсюда вытаскивать информацию для расчета, возможно, по обороту)

Итого: нам нужно получить сводную таблицу, которая даст понимание о том, какая итоговая чистая прибыль у нас есть по 5 товарам

<hr>

Порядок действий:
- получаем данные из таблиц
- преобразуем для объединения
- объединяем
- фильтруем по заданным позициям
- группируем в сводную таблицу
- производим расчёты по рентабильности
- расшифровываем данные таблицы
- записываем сводную таблицу в файл

In [63]:
import pandas as pd
import warnings

In [64]:
# снимаем ограничение на количество столбцов
pd.set_option('display.max_columns', None)

# снимаем ограничение на ширину столбцов
pd.set_option('display.max_colwidth', None)

# игнорируем предупреждения
warnings.filterwarnings(action='ignore')

In [65]:
# читаем данные из таблицы аспо деталиазция.xlsx
wb = pd.read_excel('аспо деталиазция.xlsx', sheet_name='Sheet1')
wb['Артикул поставщика'] = wb['Артикул поставщика'].astype('string')

In [66]:
wb.head(2)

№  Номер поставки           Предмет  Код номенклатуры     Бренд  \
0  15352716047        12137010  Брюки спортивные          45870607  ASPOLIFE   
1  15352716060        10833190            Куртки          16513941  ASPOLIFE   

  Артикул поставщика  \
0             444444   
1              10011   

                                                      Название Размер  \
0  Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка      M   
1       Мотокуртка мужская Моточерепаха мужская Мотоэкипировка     XL   

          Баркод Тип документа Обоснование для оплаты Дата заказа покупателем  \
0  2012056595001       Продажа              Логистика              2023-07-09   
1  2000670701045       Продажа              Логистика              2023-07-10   

  Дата продажи  Кол-во  Цена розничная  Вайлдберриз реализовал Товар (Пр)  \
0   2023-07-12       0               0                                0.0   
1   2023-07-11       0               0                                0.0   

   Согласованный продуктовый дисконт, %  Промокод %  \
0                                     0           0   
1                                     0           0   

   Итоговая согласованная скидка  \
0                              0   
1                              0   

   Цена розничная с учетом согласованной скидки  \
0                                           0.0   
1                                           0.0   

   Размер снижения кВВ из-за рейтинга, %  Размер снижения кВВ из-за акции, %  \
0                                    NaN                                 NaN   
1                                    NaN                                 NaN   

   Скидка постоянного Покупателя (СПП)  Размер кВВ, %  \
0                                  0.0            0.0   
1                                  0.0            0.0   

   Размер  кВВ без НДС, % Базовый  Итоговый кВВ без НДС, %  \
0                             0.0                      0.0   
1                             0.0                      0.0   

   Вознаграждение с продаж до вычета услуг поверенного, без НДС  \
0                                                           0.0   
1                                                           0.0   

   Возмещение за выдачу и возврат товаров на ПВЗ  \
0                                            0.0   
1                                            0.0   

   Возмещение издержек по эквайрингу  \
0                                0.0   
1                                0.0   

   Вознаграждение Вайлдберриз (ВВ), без НДС  НДС с Вознаграждения Вайлдберриз  \
0                                       0.0                               0.0   
1                                       0.0                               0.0   

   К перечислению Продавцу за реализованный Товар  Количество доставок  \
0                                             0.0                    1   
1                                             0.0                    1   

   Количество возврата  Услуги по доставке товара покупателю  \
0                    0                                  72.5   
1                    0                                 262.5   

   Общая сумма штрафов  Доплаты Виды логистики, штрафов и доплат  Стикер МП  \
0                  0.0        0            К клиенту при продаже        NaN   
1                  0.0        0             К клиенту при отмене        NaN   

  Наименование банка-эквайера  Номер офиса Наименование офиса доставки  \
0                         NaN     151362.0                         NaN   
1                         NaN          0.0      Пункт самовывоза (ПВЗ)   

   ИНН партнера                 Партнер     Склад  Страна Тип коробов  \
0  2.317088e+11  ИП Мухаметнурова Л. В.  Коледино  Россия        Моно   
1           NaN                     NaN  Коледино  Россия        Микс   

   Номер таможенной декларации  Код маркировки           ШК  Rid  \
0                          NaN             NaN  10816853936  0.0   
1                

Для суммирования данных нам необходимо поменять положительные на отрицательные и наоборот значения в таблице по некоторым колонкам при условии 'Тип документа' == 'Возврат':

In [67]:
# К перечислению за товар
wb.loc[wb['Тип документа'] == 'Возврат', 'Вайлдберриз реализовал Товар (Пр)'] = -wb['Вайлдберриз реализовал Товар (Пр)']

In [68]:
# комисся вайлдберис, НДС комиссии
wb.loc[wb['Тип документа'] == 'Возврат', 'Вознаграждение Вайлдберриз (ВВ), без НДС'] = -wb['Вознаграждение Вайлдберриз (ВВ), без НДС']
wb.loc[wb['Тип документа'] == 'Возврат', 'НДС с Вознаграждения Вайлдберриз'] = -wb['НДС с Вознаграждения Вайлдберриз']

In [69]:
# Возмещение расходов поверенного 
wb.loc[wb['Тип документа'] == 'Возврат', 'Возмещение за выдачу и возврат товаров на ПВЗ'] = -wb['Возмещение за выдачу и возврат товаров на ПВЗ']
wb.loc[wb['Тип документа'] == 'Возврат', 'Возмещение издержек по эквайрингу'] = -wb['Возмещение издержек по эквайрингу']

<hr>

### Добавление данных о хранении  
Оплату за хранение мы знаем только итоговой суммой.  
Мы будем делить сумму стоимости хранения на долю стоимости каждого товара от выручки.

In [70]:
# сумма стоимости размещения всех товаров за указанный период
warehousing = 17943.48

# сумма всей выручки
revenue = wb[(wb['Тип документа'] == 'Продажа') & (wb['Обоснование для оплаты'] == 'Продажа')] \
    ['Вайлдберриз реализовал Товар (Пр)'].sum()

# доля хранения от выручки
storage = warehousing / revenue

# добавляем колонку
wb['Стоимость хранения'] = 0

# добавляем в таблицу данные о хранении только в строки о продаже
wb.loc[(wb['Тип документа'] == 'Продажа') & (wb['Обоснование для оплаты'] == 'Продажа'), 'Стоимость хранения'] = wb['Вайлдберриз реализовал Товар (Пр)'] * storage

<hr>

### Добавление данных о рекламе
Название рекламных компаний нам ничего не говорят о конкретном товаре, который мы рекламируем.  
Для расчета мы можем только приблизительно понять к какой категории относится рекламная компания.  
После этого мы суммируем рекламные расходы по категориям, а затем разбиваем эти суммы на каждый товар в зависимости от категории, аналогичным способом Стоимости хранения.

In [71]:
# создаём столбец с данными о рекламе
wb['Реклама'] = 0

In [72]:
# получаем данные из таблиц о рекламных компаниях
ad_1 = pd.read_excel('Расходы на рекламу.xlsx', sheet_name='Карточка товара')
ad_2 = pd.read_excel('Расходы на рекламу.xlsx', sheet_name='Каталог')
ad_3 = pd.read_excel('Расходы на рекламу.xlsx', sheet_name='Поиск')

ad_1

Id кампании                                 Кампания  Просмотры  Клики  \
0       7735213     Тактические в карточке товара 6 мето       8869     69   
1       7230522      Шлем черный карточка товара 2 место      39480    355   
2       5100067       Черные перчатки - карточка 4 место       4463     64   
3       7933298               Очки красные мото карточка       9831     63   
4       5124006        Красные - карточка товара 2 место      10790     95   
5       7263164                      Защита тела - шлема       2744     32   
6       5124049  Все цвета мотоперчатки карточка 3 место       3801     31   
7       7263174                      Защита тела - штаны       1234     33   
8       5124020        Зеленые карточка товара - 5 место      12313    106   
9       7966828                     Наколенники карточка       6666     98   
10      6983390                 мотоочки черные карточка      33497    278   
11      7230545                     Защита тела - куртки       2910     39   
12      7263819                      Тент - карточка 4xl        791     26   
13      6694865           Мотошлем фокс -карточка товара      29720    276   
14      2807430                            Тент Карточка        592     13   

    Добавления в корзину  Заказы  Сумма, р               Предмет  
0                      4       1      2500       Очки спортивные  
1                     10       3      4500      Шлемы спортивные  
2                      6       1      1245              Перчатки  
3                      4       0      1174       Очки спортивные  
4                      5       2      3500       Очки спортивные  
5                      1       0       111      Шлемы спортивные  
6                      0       0      1230              Перчатки  
7                      3       0        43      Брюки спортивные  
8                      6       0      3500           Наколенники  
9                     12       5      1500           Наколенники  
10                    23       1      3605       Очки спортивные  
11                     2       0       141                Куртки  
12                     4       2        54  Тенты для мотоциклов  
13                     9       0      3210      Шлемы спортивные  
14                     4       1        36  Тенты для мотоциклов

In [73]:
# объединяем данные с разных типов рекламы и группируем по категориям товаров
advertising = pd.concat([ad_1[['Предмет', 'Сумма, р']], ad_2[['Предмет', 'Сумма, р']], ad_3[['Предмет', 'Сумма, р']]])
advertising = advertising.groupby('Предмет').agg({'Сумма, р': sum}).reset_index()
advertising

Предмет  Сумма, р
0      Брюки спортивные        43
1                Куртки      2003
2           Наколенники      6610
3       Очки спортивные     17045
4              Перчатки     11610
5  Тенты для мотоциклов      1386
6      Шлемы спортивные     16951

In [74]:
# добавим рекламные расходы к каждому товару в общую таблицу
for item in advertising['Предмет'].tolist():
    revenue = wb[(wb['Предмет'] == item) & (wb['Тип документа'] == 'Продажа') & (wb['Обоснование для оплаты'] == 'Продажа')] \
        ['Вайлдберриз реализовал Товар (Пр)'].sum()
    warehousing = advertising[advertising['Предмет'] == item]['Сумма, р'].values[0]
    storage = warehousing / revenue
    wb.loc[(wb['Предмет'] == item) & (wb['Тип документа'] == 'Продажа') & (wb['Обоснование для оплаты'] == 'Продажа'), 'Реклама'] = wb['Вайлдберриз реализовал Товар (Пр)'] * storage

<hr>

### Добавим данных о себестимости товара, и стоимости логистики до ваилдберис

Мы могли бы обратится к таблице "Фин модель" но оказалось что она не содержит точных данных об артикулах товаров с которыми можно было бы сопоставить себестоимость товара и стоимость доставки до ваилдберис.

Пришлось наугад примерно выбрать подходящие названиям модели, дописать к нам артикулы товаров и пересобрать данные в таблицу 'Фин_модель_ред.xlsx'

После этого мы можем добавить эту информацию к общим данным.

In [75]:
# получим данные о рекламе
finmod = pd.read_excel('Фин_модель_ред.xlsx')[['Артикул поставщика', 'Себ-ть товара на нашем складе', 'Доставка до WB']]
finmod['Артикул поставщика'] = finmod['Артикул поставщика'].astype('string')
finmod

Артикул поставщика  Себ-ть товара на нашем складе  Доставка до WB
0                  10010                    2120.029957       40.000000
1                  10011                    1895.418397       40.000000
2               00016356                     347.276969        4.291408
3            00016336633                     374.176267        4.291408
4               00016465                     396.772708        4.291408
5                 567003                    1115.446900       40.000000
6  shlem_zashchitnye_red                    2728.618784       18.061594
7             motoshtany                    1257.792971       17.000000
8                 444444                    1329.476943       17.000000
9          shtany_nabory                    1339.161271       17.000000

In [76]:
# добавим данные о себестоимости товара и стоимости доставки
wb = wb.merge(finmod, on='Артикул поставщика', how='left')

In [77]:
# удалим лишнее, в сучае если это не продажи и возвраты
wb.loc[(wb['Тип документа'] == 'Продажа') & (wb['Обоснование для оплаты'] != 'Продажа') | \
    (wb['Тип документа'] == 'Возврат') & (~wb['Обоснование для оплаты'].isin(['Возврат', 'Авансовая оплата за товар без движения'])), \
       'Себ-ть товара на нашем складе'] = 0

wb.loc[(wb['Тип документа'] == 'Продажа') & (wb['Обоснование для оплаты'] != 'Продажа') | \
    (wb['Тип документа'] == 'Возврат') & (~wb['Обоснование для оплаты'].isin(['Возврат', 'Авансовая оплата за товар без движения'])), \
       'Доставка до WB'] = 0

In [78]:
wb

№  Номер поставки               Предмет  Код номенклатуры  \
0     15352716047        12137010      Брюки спортивные          45870607   
1     15352716060        10833190                Куртки          16513941   
2     15352716062        10833190              Перчатки         106215467   
3     15352716065        12137010      Брюки спортивные          45870607   
4     15352716067        11836225              Перчатки          35296649   
...           ...             ...                   ...               ...   
1634  15352719291        12832901      Брюки спортивные          45870607   
1635  15352719293        13027613  Тенты для мотоциклов         158136060   
1636  15352719294        12823021       Защитные наборы         163363358   
1637  15352719296        12823021      Брюки спортивные          45870607   
1638  15352719298        12823021      Брюки спортивные          45870607   

         Бренд Артикул поставщика  \
0     ASPOLIFE             444444   
1     ASPOLIFE              10011   
2     ASPOLIFE               5654   
3     ASPOLIFE             444444   
4     ASPOLIFE           00016465   
...        ...                ...   
1634  ASPOLIFE             444444   
1635  ASPOLIFE             567004   
1636  ASPOLIFE      shtany_nabory   
1637  ASPOLIFE             444444   
1638  ASPOLIFE             444444   

                                                         Название Размер  \
0     Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка      M   
1          Мотокуртка мужская Моточерепаха мужская Мотоэкипировка     XL   
2      Тактические перчатки мотоперчатки военные защитные для СВО      M   
3     Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка      M   
4               Мотоперчатки мужские Мотоэкипировка мото перчатки      M   
...                                                           ...    ...   
1634  Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка      M   
1635    Чехол для мотоцикла Тент для мотоцикла Защитный Моточехол      0   
1636     Мотоштаны с защитой Штаны для мотокросса Мотозащита Мото     XL   
1637  Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка    2XL   
1638  Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка    2XL   

             Баркод Тип документа            Обоснование для оплаты  \
0     2012056595001       Продажа                         Логистика   
1     2000670701045       Продажа                         Логистика   
2     2036898025038       Продажа                         Логистика   
3     2012056595001       Продажа                           Продажа   
4     2003287108009       Продажа                           Продажа   
...             ...           ...                               ...   
1634  2012056595001       Продажа  Возмещение издержек по перевозке   
1635  2037732531197       Продажа  Возмещение издержек по перевозке   
1636  2037921160450       Продажа  Возмещение издержек по перевозке   
1637  2012056595032       Продажа  Возмещение издержек по перевозке   
1638  2012056595032       Продажа  Возмещение издержек по перевозке   

     Дата заказа покупателем Дата продажи  Кол-во  Цена розничная  \
0                 2023-07-09   2023-07-12       0               0   
1                 2023-07-10   2023-07-11       0               0   
2                 2023-07-09   2023-07-11       0               0   
3                 2023-07-10   2023-07-11       1           22271   
4                 2023-07-07   2023-07-11       1            4779   
...                      ...          ...     ...             ...   
1634                     NaN          NaN       1               0   
1635                     NaN          NaN       1               0   
1636                     NaN          NaN       1               0   
1637                     NaN          NaN       1               0   
1638                     NaN          NaN       1               0   

      Вайлдберриз реализовал Товар (Пр)  Согласован

<hr>

### Сводная таблица

После того как мы собрали и объединили все данные мы можем:
- отобрать только интересующие нас позиции
- сгруппировать данные по каждому товару за веь период
- посчитать что нам перечислит ваилдберис в колонку 'Итого к перечислению продавцу'
- посчитать чистую прибыль

In [79]:
# оставим только необходимые для расчёта колонки
wb = wb[['Код номенклатуры', 
         'Артикул поставщика',
         'Предмет',
         'Название', 
         'Тип документа', 
         'Кол-во',
         'Обоснование для оплаты', 
         'Вайлдберриз реализовал Товар (Пр)', 
         'Вознаграждение Вайлдберриз (ВВ), без НДС', 
         'НДС с Вознаграждения Вайлдберриз', 
         'Услуги по доставке товара покупателю',
         'Возмещение за выдачу и возврат товаров на ПВЗ', 
         'Возмещение издержек по эквайрингу', 
         'Общая сумма штрафов',
         'Виды логистики, штрафов и доплат', 
         'Стоимость хранения',
         'Реклама',
         'ШК', 
         'Себ-ть товара на нашем складе', 
         'Доставка до WB']]

In [80]:
# определяем товары, которые нас интересуют
goods_types = ['Мотоперчатки', 'Мотоштаны', 'Мотокуртка', 'Мотошлем', 'Мототент']

# оставляем только выбрнные товары
wb['categ'] = wb['Название'].apply(lambda x: x.split(' ')[0])
wb = wb[wb.categ.isin(goods_types)]

# также удаляем тактические перчатки
wb = wb[wb['Код номенклатуры'] != 35302428]

# Будем работать по этим товарам
wb.sort_values('Название')['Название'].unique()

array(['Мотокуртка защитная для мотоцикла мото черепаха',
       'Мотокуртка мужская Моточерепаха мужская Мотоэкипировка',
       'Мотоперчатки Мотоэкипировка мужская мото перчатки летние',
       'Мотоперчатки Перчатки для мотоцикла Мотоэкипировка',
       'Мотоперчатки мужские Мотоэкипировка мото перчатки',
       'Мототент для мотоцикла Тент чехол на мотоцикл Защитный',
       'Мотошлем кроссовый Мотозащита Шлем для мотоцикла Эндуро',
       'Мотоштаны мужские fox Штаны для мотокросса Мотобрюки фокс',
       'Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка',
       'Мотоштаны с защитой Штаны для мотокросса Мотозащита Мото'],
      dtype=object)

In [81]:
# сгруппируем данные по выбранным товарам
group = wb.groupby('Название').agg({'Артикул поставщика': 'first', 
                                    'Кол-во': sum, 
                                    'Вайлдберриз реализовал Товар (Пр)': sum, 
                                    'Вознаграждение Вайлдберриз (ВВ), без НДС': sum, 
                                    'НДС с Вознаграждения Вайлдберриз': sum, 
                                    'Услуги по доставке товара покупателю': sum,
                                    'Возмещение за выдачу и возврат товаров на ПВЗ': sum, 
                                    'Возмещение издержек по эквайрингу': sum, 
                                    'Общая сумма штрафов': sum, 
                                    'Стоимость хранения': sum, 
                                    'Реклама': sum,
                                    'Себ-ть товара на нашем складе': sum, 
                                    'Доставка до WB': sum})
group.reset_index(inplace= True)

In [82]:
# считаем Итого к перчислению продавцу
group['Итого к перечислению продавцу'] = group['Вайлдберриз реализовал Товар (Пр)'] \
    - group['Вознаграждение Вайлдберриз (ВВ), без НДС'] \
    - group['НДС с Вознаграждения Вайлдберриз'] \
    - group['Услуги по доставке товара покупателю'] \
    - group['Стоимость хранения'] \
    - group['Возмещение за выдачу и возврат товаров на ПВЗ'] \
    - group['Возмещение издержек по эквайрингу'] \
    - group['Общая сумма штрафов'] \
    - group['Реклама']

# считаем все затраты
group['Все затраты'] = group['Вознаграждение Вайлдберриз (ВВ), без НДС'] \
    + group['НДС с Вознаграждения Вайлдберриз'] \
    + group['Услуги по доставке товара покупателю'] \
    + group['Стоимость хранения'] \
    + group['Возмещение за выдачу и возврат товаров на ПВЗ'] \
    + group['Возмещение издержек по эквайрингу'] \
    + group['Общая сумма штрафов'] \
    + group['Реклама'] \
    + group['Себ-ть товара на нашем складе'] \
    + group['Доставка до WB']

# прибыль
group['Прибыль'] = (group['Вайлдберриз реализовал Товар (Пр)'] - group['Все затраты'])

# считаем ROI
group['ROI'] = round(group['Прибыль'] / group['Все затраты'], 2)

In [83]:
# переставляем колонки
group = group[['Артикул поставщика', 
        'Название',
       'Кол-во', 
        'Вайлдберриз реализовал Товар (Пр)', 
       'Вознаграждение Вайлдберриз (ВВ), без НДС', 
       'НДС с Вознаграждения Вайлдберриз', 
       'Услуги по доставке товара покупателю', 
       'Стоимость хранения', 
       'Возмещение за выдачу и возврат товаров на ПВЗ', 
       'Возмещение издержек по эквайрингу', 
       'Общая сумма штрафов', 
       'Реклама', 
        'Итого к перечислению продавцу', 
               
       'Себ-ть товара на нашем складе', 
       'Доставка до WB', 
    
       'Все затраты',
       'Прибыль',
       'ROI']]

group['Стоимость хранения'] = round(group['Стоимость хранения'], 2)
group['Реклама'] = round(group['Реклама'], 2)
group['Итого к перечислению продавцу'] = round(group['Итого к перечислению продавцу'], 2)
group['Себ-ть товара на нашем складе'] = round(group['Себ-ть товара на нашем складе'], 2)
group['Доставка до WB'] = round(group['Доставка до WB'], 2)
group['Вайлдберриз реализовал Товар (Пр)'] = round(group['Вайлдберриз реализовал Товар (Пр)'], 2)
group['Все затраты'] = round(group['Все затраты'], 2)
group['Прибыль'] = round(group['Прибыль'], 2)

group

Артикул поставщика  \
0                  10010   
1                  10011   
2               00016356   
3            00016336633   
4               00016465   
5                 567003   
6  shlem_zashchitnye_red   
7             motoshtany   
8                 444444   
9          shtany_nabory   

                                                      Название  Кол-во  \
0              Мотокуртка защитная для мотоцикла мото черепаха      14   
1       Мотокуртка мужская Моточерепаха мужская Мотоэкипировка      60   
2     Мотоперчатки Мотоэкипировка мужская мото перчатки летние      70   
3           Мотоперчатки Перчатки для мотоцикла Мотоэкипировка      35   
4            Мотоперчатки мужские Мотоэкипировка мото перчатки      29   
5       Мототент для мотоцикла Тент чехол на мотоцикл Защитный      18   
6      Мотошлем кроссовый Мотозащита Шлем для мотоцикла Эндуро       0   
7    Мотоштаны мужские fox Штаны для мотокросса Мотобрюки фокс      13   
8  Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка      96   
9     Мотоштаны с защитой Штаны для мотокросса Мотозащита Мото       1   

   Вайлдберриз реализовал Товар (Пр)  \
0                           10458.00   
1                           87088.26   
2                           28825.00   
3                           16276.00   
4                           13076.00   
5                           12329.00   
6                               0.00   
7                           12487.00   
8                          146063.00   
9                               0.00   

   Вознаграждение Вайлдберриз (ВВ), без НДС  НДС с Вознаграждения Вайлдберриз  \
0                                   -986.85                           -197.37   
1                                  -7095.62                          -1419.14   
2                                  -1548.03                           -309.65   
3                                   -878.62                           -175.73   
4                                   -652.51                           -130.51   
5                                   -565.61                           -113.14   
6                                      0.00                              0.00   
7                                     21.90                              4.38   
8                                  -7550.26                          -1510.10   
9                                    -19.59                             -3.92   

   Услуги по доставке товара покупателю  Стоимость хранения  \
0                               2485.00              319.55   
1                               8028.75             1695.75   
2                               3170.00              519.57   
3                               1857.50              293.37   
4                               1187.50              235.69   
5                               1131.55              297.36   
6                                  0.00                0.00   
7                                595.00              333.55   
8                               6680.00             2632.77   
9                                  0.00                0.00   

   Возмещение за выдачу и возврат товаров на ПВЗ  \
0                                         215.13   
1                                        1532.30   
2                                         421.54   
3                                         311.59   
4                                         201.84   
5                                         111.47   
6                                           0.00   
7                                           0.00   
8                                        2374.49   
9                                           0.00   

   Возмещение издержек по эквайрингу  Общая сумма штрафов  Реклама  \
0                              59.64                 0.00   317.60   
1                             731.25                 0.00  1685.40   
2                             170.75                 0.00  3151.54   
3 

In [84]:
group.drop(columns='Название').sum()

Кол-во                                              336.00
Вайлдберриз реализовал Товар (Пр)                326602.26
Вознаграждение Вайлдберриз (ВВ), без НДС         -19275.19
НДС с Вознаграждения Вайлдберриз                  -3855.18
Услуги по доставке товара покупателю              25135.30
Стоимость хранения                                 6327.61
Возмещение за выдачу и возврат товаров на ПВЗ      5168.36
Возмещение издержек по эквайрингу                  2162.50
Общая сумма штрафов                                2473.67
Реклама                                            8694.98
Итого к перечислению продавцу                    299770.20
Себ-ть товара на нашем складе                    180606.06
Доставка до WB                                     3073.85
Все затраты                                      210511.96
Прибыль                                          116090.30
ROI                                                   1.54
dtype: float64

### Расшифровка:

Артикул поставщика  
Название (товара)  
Кол-во (количество проданных товаров) 336  
Вайлдберриз реализовал Товар (Пр)(сумма по чекам покупателей) 326602.26  
Вознаграждение Вайлдберриз (ВВ), без НДС(комиссия ВБ) -19275.19  
НДС с Вознаграждения Вайлдберриз(НДС комиссии ВБ) -3855.18  
Услуги по доставке товара покупателю(логистика ВБ) 25135.30  
Стоимость хранения(хранение ВБ) 6327.61  
Возмещение за выдачу и возврат товаров на ПВЗ(услуги ПВЗ) 5168.36  
Возмещение издержек по эквайрингу(эквайринг ПВЗ) 2162.50  
Общая сумма штрафов 2473.67  
Реклама 8694.97  

Итого к перечислению продавцу(от вайлдберис после вычета всех расходов) 299770.21  

Себ-ть товара на нашем складе 180606.05  
Доставка до WB 3073.85  

Все затраты (комиссии и НДС ВБ, логистика ВБ, хранение ВБ, услуги и эквайринг ПВЗ, штрафы, реклама, себестоимость и доставка до ВБ) 210511.96  
Прибыль(после вычета всех затрат) 116090.29  

In [85]:
# Основной вопрос: Сколько заработано чистой прибыли по следующим товарам?
group[['Артикул поставщика', 'Название', 'Кол-во', 'Прибыль']]

Артикул поставщика  \
0                  10010   
1                  10011   
2               00016356   
3            00016336633   
4               00016465   
5                 567003   
6  shlem_zashchitnye_red   
7             motoshtany   
8                 444444   
9          shtany_nabory   

                                                      Название  Кол-во  \
0              Мотокуртка защитная для мотоцикла мото черепаха      14   
1       Мотокуртка мужская Моточерепаха мужская Мотоэкипировка      60   
2     Мотоперчатки Мотоэкипировка мужская мото перчатки летние      70   
3           Мотоперчатки Перчатки для мотоцикла Мотоэкипировка      35   
4            Мотоперчатки мужские Мотоэкипировка мото перчатки      29   
5       Мототент для мотоцикла Тент чехол на мотоцикл Защитный      18   
6      Мотошлем кроссовый Мотозащита Шлем для мотоцикла Эндуро       0   
7    Мотоштаны мужские fox Штаны для мотокросса Мотобрюки фокс      13   
8  Мотоштаны мужские Штаны для мотокросса Мотобрюки Экипировка      96   
9     Мотоштаны с защитой Штаны для мотокросса Мотозащита Мото       1   

    Прибыль  
0  -6874.90  
1  27737.85  
2   9538.12  
3   5450.44  
4   4303.65  
5   -472.35  
6  -2473.67  
7   6359.49  
8  72498.16  
9     23.51

In [86]:
# Итого: нам нужно получить сводную таблицу, которая даст понимание о том, какая итоговая чистая прибыль у нас есть по 5 товарам
# (запишем результат в таблицу)
group.to_excel("test_vova_sizov.xlsx", sheet_name='pivot table', index=False)